pink/tasks/relative_frame_task.py

```diff
@@ -0,0 +1,238 @@
+            gain: Task gain :math:`\alpha \in [0, 1]` for additional low-pass
+                filtering. Defaults to 1.0 (no filtering) for dead-beat
+                control.
+        """
+        super().__init__(
+            cost=np.ones(6),  # updated below
+            gain=gain,
+            lm_damping=lm_damping,
+        )
+        self.frame = frame
+        self.root = root
+        self.lm_damping = lm_damping
+        self.transform_target_to_root = None
+        #
+        self.set_position_cost(position_cost)
+        self.set_orientation_cost(orientation_cost)
+
+    def set_position_cost(
+        self, position_cost: Union[float, Sequence[float], np.ndarray]
+    ) -> None:
+        r"""Set a new cost for all 3D position coordinates.
+
+        Args:
+            position_cost: Contribution of position errors to the normalized
+                cost, in :math:`[\mathrm{cost}] / [\mathrm{m}]`. If this is a
+                vector, the cost is anisotropic and each coordinate corresponds
+                to an axis of the frame.
+        """
+        if isinstance(position_cost, float):
+            assert position_cost >= 0.0
+        else:  # not isinstance(position_cost, float)
+            assert all(cost >= 0.0 for cost in position_cost)
+        if isinstance(self.cost, np.ndarray):
+            self.cost[0:3] = position_cost
+        else:  # self.cost is not a vector
+            raise TaskDefinitionError(
+                "Frame task cost should be a vector, "
+                f"currently cost={self.cost}"
+            )
+
+    def set_orientation_cost(
+        self, orientation_cost: Union[float, Sequence[float], np.ndarray]
+    ) -> None:
+        r"""Set a new cost for all 3D orientation coordinates.
+
+        Args:
+            orientation_cost: Contribution of orientation errors to the
+                normalized cost, in :math:`[\mathrm{cost}] / [\mathrm{rad}]`.
+                If this is a vector, the cost is anisotropic and each
+                coordinate corresponds to an axis of the frame.
+        """
+        if isinstance(orientation_cost, float):
+            assert orientation_cost >= 0.0
+        else:  # not isinstance(orientation_cost, float)
+            assert all(cost >= 0.0 for cost in orientation_cost)
+        if isinstance(self.cost, np.ndarray):
+            self.cost[3:6] = orientation_cost
+        else:  # self.cost is not a vector
+            raise TaskDefinitionError(
+                "Frame task cost should be a vector, "
+                f"currently cost={self.cost}"
+            )
+
+    def set_target(
+        self,
+        transform_target_to_root: pin.SE3,
+    ) -> None:
+        """Set task target pose in the root frame.
+
+        Args:
+            transform_target_to_root: Affine transform matrix (4×4) from the
+                task target frame to the root frame.
+        """
+        self.transform_target_to_root = transform_target_to_root.copy()
+
```

Comment:
This makes sense in internal calculations, but from an API standpoint I think it's better in this direction for consistency with the `FrameTask`:

- Frame task has "transform from target to world" (because common practice is to work with $T_{0b}$ rather than $T_{b0}$)
- Relative frame task has "transform from target to root"